packages/transport/src/transports/abstract.ts

```diff
@@ -0,0 +1,408 @@
+import * as protobuf from 'protobufjs/light';
+
+import { Deferred } from '@trezor/utils';
+import {
+    MessageFromTrezor,
+    Session,
+    Descriptor,
+    AsyncResultWithTypedError,
+    ResultWithTypedError,
+    Success,
+    AnyError,
+} from '../types';
+import { TypedEmitter } from '../types/typed-emitter';
+import { getAbortController } from '../utils/abortController';
+import { success, error, unknownError } from '../utils/result';
+
+import * as SESSION_ERRORS from '../sessions/errors';
+import * as TRANSPORT_ERRORS from './errors';
+import * as COMMON_ERRORS from '../errors';
+import * as INTERFACE_ERRORS from '../interfaces/errors';
+
+export type AcquireInput = {
+    path: string;
+    previous?: Session;
+};
+
+type DeviceDescriptorDiff = {
+    didUpdate: boolean;
+    descriptors: Descriptor[];
+    connected: Descriptor[];
+    disconnected: Descriptor[];
+    changedSessions: Descriptor[];
+    acquired: Descriptor[];
+    acquiredByMyself: Descriptor[];
+    acquiredElsewhere: Descriptor[];
+    released: Descriptor[];
+    releasedByMyself: Descriptor[];
+    releasedElsewhere: Descriptor[];
+};
+
+export const TRANSPORT = {
+    START: 'transport-start',
+    ERROR: 'transport-error',
+    UPDATE: 'transport-update',
+    DISABLE_WEBUSB: 'transport-disable_webusb',
+    REQUEST_DEVICE: 'transport-request_device',
+} as const;
+
+type ConstructorParams = {
+    messages: Record<string, any>;
+    signal?: AbortSignal;
+};
+
+export abstract class Transport extends TypedEmitter<{
+    [TRANSPORT.UPDATE]: DeviceDescriptorDiff;
+    // todo: transport error type better
+    [TRANSPORT.ERROR]: string;
+}> {
+    public name: 'BridgeTransport' | 'NodeUsbTransport' | 'WebUsbTransport' | 'AbstractTransport';
```

Comment:
Is 'AbstractTransport' a transport as well? Does it make sense to have it here listed?

Reply:
good idea, lets get rid of it 33cde4b18faf97c67bb48078c8030106a31882fb